src/core/namespace.c

```diff
@@ -687,7 +711,19 @@ static int mount_private_dev(MountEntry *m) {
         (void) symlink("/run/systemd/journal/dev-log", devlog);
 
         NULSTR_FOREACH(d, devnodes) {
-                r = clone_device_node(d, temporary_mount);
+                if (can_mknod) {
+                        r = clone_device_node(d, temporary_mount, can_mknod);
+                        if (r != -EPERM) {
+                                if (errno != EPERM)
+                                        goto fail;
+
+                                can_mknod = false;
+                        } else {
+                                continue;
+                        }
```

Comment:
coding style nitpick: no {} around single-line if blocks

Reply:
actually, the "continue" branch is unnecessary anyway, no? it's the last thing we do in the loop iteration anyway...